src/tests/suite/daos_rebuild.c

```diff
@@ -333,1467 +328,208 @@ rebuild_io_validate(test_arg_t *arg, daos_obj_id_t *oids, int oids_nr,
-}
-
-static void
-rebuild_drop_obj(void **state)
-{
-	test_arg_t	*arg = *state;
-	daos_obj_id_t	oids[OBJ_NR];
-	int		i;
-
-	if (!test_runable(arg, 6))
-		return;
+	if (arg->myrank == 0) {
+		print_message("daos_cont_open()\n");
+		rc = daos_cont_open(arg->pool.poh, arg->co_uuid, DAOS_COO_RW,
+				    &arg->coh, &arg->co_info, NULL);
+		if (rc)
+			print_message("daos_cont_open failed, rc: %d\n", rc);
 
-	for (i = 0; i < OBJ_NR; i++) {
-		oids[i] = dts_oid_gen(DAOS_OC_R3S_SPEC_RANK, 0, arg->myrank);
-		oids[i] = dts_oid_set_rank(oids[i], ranks_to_kill[0]);
+		print_message("container open "DF_UUIDF"\n",
+			       DP_UUID(arg->co_uuid));
 	}
-
-	rebuild_io(arg, oids, OBJ_NR);
-
-	/* Set drop REBUILD_OBJECTS reply on all servers */
-	if (arg->myrank == 0)
-		daos_mgmt_set_params(arg->group, 0, DSS_KEY_FAIL_LOC,
-				     DAOS_REBUILD_DROP_OBJ | DAOS_FAIL_ONCE,
-				     0, NULL);
 	MPI_Barrier(MPI_COMM_WORLD);
-	rebuild_single_pool_rank(arg, ranks_to_kill[0]);
-
-	rebuild_io_validate(arg, oids, OBJ_NR, true);
-}
-
-static void
-rebuild_update_failed(void **state)
-{
-	test_arg_t	*arg = *state;
-	daos_obj_id_t	oids[OBJ_NR];
-	int		i;
-
-	if (!test_runable(arg, 6))
-		return;
+	if (arg->multi_rank)
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+	if (rc)
+		return rc;
 
-	for (i = 0; i < OBJ_NR; i++) {
-		oids[i] = dts_oid_gen(DAOS_OC_R3S_SPEC_RANK, 0, arg->myrank);
-		oids[i] = dts_oid_set_rank(oids[i], ranks_to_kill[0]);
-		oids[i] = dts_oid_set_tgt(oids[i], DEFAULT_FAIL_TGT);
+	/** broadcast container info */
+	if (arg->multi_rank) {
+		MPI_Bcast(&rc, 1, MPI_INT, 0, MPI_COMM_WORLD);
+		handle_share(&arg->coh, HANDLE_CO, arg->myrank, arg->pool.poh,
+			     0);
 	}
 
-	rebuild_io(arg, oids, OBJ_NR);
-
-	/* Set drop scan reply on all servers */
-	if (arg->myrank == 0)
-		daos_mgmt_set_params(arg->group, 0, DSS_KEY_FAIL_LOC,
-				     DAOS_REBUILD_UPDATE_FAIL | DAOS_FAIL_ONCE,
-				     0, NULL);
-	MPI_Barrier(MPI_COMM_WORLD);
-	rebuild_single_pool_target(arg, ranks_to_kill[0], DEFAULT_FAIL_TGT);
+	return 0;
 }
 
 static void
-rebuild_multiple_pools(void **state)
+rebuild_offline(void **state)
 {
 	test_arg_t	*arg = *state;
-	test_arg_t	*args[2] = { 0 };
 	daos_obj_id_t	oids[OBJ_NR];
 	int		i;
-	int		rc;
-
-	if (!test_runable(arg, 6))
-		return;
+	unsigned int	required_nodes = 3;
 
-	args[0] = arg;
-	/* create/connect another pool */
-	rc = test_setup((void **)&args[1], SETUP_CONT_CONNECT, arg->multi_rank,
-			REBUILD_POOL_SIZE, NULL);
-	if (rc) {
-		print_message("open/connect another pool failed: rc %d\n", rc);
+	print_message("rebuild_offline test - %u server nodes, 2-way replica\n",
+			required_nodes);
+	if (!test_runable(arg, required_nodes))
 		return;
-	}
 
 	for (i = 0; i < OBJ_NR; i++) {
-		oids[i] = dts_oid_gen(DAOS_OC_R3S_SPEC_RANK, 0, arg->myrank);
+		oids[i] = dts_oid_gen(DAOS_OC_R2S_SPEC_RANK, 0, arg->myrank);
+		print_message("rank %d set for oid:"DF_U64"."DF_U64"\n",
+				ranks_to_kill[0], oids[i].hi, oids[i].lo);
```

Comment:
you should print oid after oid_set_rank(), since set_rank will change the oid. I assume you use wrong oid(empty object) to print the layout, that is why you get the "same" layout?